src/createWidgetBase.ts

```diff
@@ -201,14 +203,34 @@ const createWidget: WidgetBaseFactory = createStateful
 
 			setProperties(this: Widget<WidgetProperties>, properties: WidgetProperties) {
 				const internalState = widgetInternalStateMap.get(this);
-				const processedProperties = this.diffProperties(internalState.previousProperties, properties);
-				internalState.properties = processedProperties.properties;
-				if (processedProperties.changedKeys.length) {
+
+				const diffPropertyResults: { [index: string]: PropertyChangeRecord } = {};
+				const diffPropertyChangedKeys: string[] = [];
+
+				internalState.diffPropertyFunctionMap.forEach((property: string, diffFunction: string) => {
+					const previousProperty = internalState.previousProperties[property];
+					const newProperty = properties[property];
+					const result: PropertyChangeRecord = this[diffFunction](previousProperty, newProperty);
+
+					if (result.changed) {
```

Comment:
even though this will be guarded by TS, I think we should check whether `result` exists?